theseus_gui/src/components/RowDisplay.vue

```diff
@@ -0,0 +1,202 @@
+import { ChevronLeftIcon, ChevronRightIcon } from 'omorphia'
+import Instance from '@/components/ui/Instance.vue'
+import News from '@/components/ui/News.vue'
+import { onMounted, onUnmounted, ref } from 'vue'
+
+const props = defineProps({
+  instances: Array,
+  news: Array,
+  label: String,
+  canPaginate: Boolean,
+})
+
+const allowPagination = ref(false)
+const modsRow = ref(null)
+const newsRow = ref(null)
+const trendingRow = ref(null)
+
+const shouldRenderNormalInstances =
+  props.instances && props.instances?.length !== 0 && props.instances?.some((i) => !i.trending)
+const shouldRenderNews = props.news && props.news?.length !== 0
+const shouldRenderTrending =
+  props.instances && props.instances?.length !== 0 && props.instances?.every((i) => i.trending)
+
+const handlePaginationDisplay = () => {
+  let parentsRow
+  if (shouldRenderNormalInstances) parentsRow = modsRow.value
+  if (shouldRenderNews) parentsRow = newsRow.value
+  else if (shouldRenderTrending) parentsRow = trendingRow.value
+
+  if (!parentsRow) return
+
+  const children = parentsRow.children
+  const lastChild = children[children.length - 1]
+  const childBox = lastChild.getBoundingClientRect()
+
+  if (childBox.x + childBox.width > window.innerWidth) allowPagination.value = true
+  else allowPagination.value = false
+}
+
+onMounted(() => {
+  if (props.canPaginate) window.addEventListener('resize', handlePaginationDisplay)
+})
+
+onUnmounted(() => {
+  if (props.canPaginate) window.removeEventListener('resize', handlePaginationDisplay)
+})
+
+const handleLeftPage = () => {
+  if (shouldRenderNormalInstances) modsRow.value.scrollLeft -= 100
+  else if (shouldRenderNews) newsRow.value.scrollLeft -= 100
+  else if (shouldRenderTrending) trendingRow.value.scrollLeft -= 100
+}
+
+const handleRightPage = () => {
+  if (shouldRenderNormalInstances) modsRow.value.scrollLeft += 100
+  else if (shouldRenderNews) newsRow.value.scrollLeft += 100
+  else if (shouldRenderTrending) trendingRow.value.scrollLeft += 100
+}
+</script>
+
+<template>
+  <div class="row">
+    <div class="header">
+      <p>{{ props.label }}</p>
+      <hr aria-hidden="true" />
+      <div v-if="allowPagination" class="pagination">
+        <ChevronLeftIcon @click="handleLeftPage" />
+        <ChevronRightIcon @click="handleRightPage" />
+      </div>
+    </div>
+    <section ref="modsRow" class="mods" v-if="shouldRenderNormalInstances">
+      <Instance
+        v-for="instance in props.instances"
+        :key="instance.id"
+        display="gallery"
+        :instance="instance"
+      />
+    </section>
+    <section ref="newsRow" class="news" v-else-if="shouldRenderNews">
+      <News v-for="news in props.news" :key="news.id" :news="news" />
+    </section>
+    <section ref="trendingRow" class="trending" v-else-if="shouldRenderTrending">
+      <Instance
+        v-for="instance in props.instances"
+        :key="instance.id"
+        display="card"
+        :instance="instance"
+      />
```

Comment:
the corners aren't rounded here (use the avatar component for the icon) and the hover state is off.

Reply:
Using the Avatar component ended up being unsensible to use here, unfortunately.